### PR TITLE
Issue 1840

### PR DESCRIPTION
**Pull request by schlenkibus**

### CODE REVIEW

File: projects/NonMaps/src/main/java/com/nonlinearlabs/client/WebSocketConnection.java [hunk 1]
```diff
@@ -56,16 +56,17 @@ private void hideConnectionError() {
...
 
 		public void onServerUpdate(String text);
 	}
 
 	ServerListener listener;
-	static String clientId = Uuid.random();
+	String clientId;
 
 	public void connectToServer(ServerListener listener) {
+		clientId = Uuid.random();
```
Comment (hhoegelo) [inline projects/NonMaps/src/main/java/com/nonlinearlabs/client/WebSocketConnection.java:69]: It is important, that a client instance uses the same clientId always, so the playground can close old web sockets tied to the same clientId. As it is now, each reconnect (e.g. in case of a connection error) will create a new web socket and keep the old one.

tl;dr: clientId shall be assigned at construction time of WebSocketConnection only.

File: projects/NonMaps/src/main/java/com/nonlinearlabs/client/integrationTests/OracleTests.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        deleteBank();
+
+        start();
+    }
+
+    private void selectPresetViaNewWebSocket() {
+        addStep(() -> {
+            webSocket.send("/presets/banks/select-preset?uuid=" + findBank().getPreset(0).getUUID().toString() + "&isOracle=true");
+        }, () -> {
+            boolean wasOracleOfServer = server.lastDocumentCouldOmitOracles();
```
Comment (hhoegelo) [inline projects/NonMaps/src/main/java/com/nonlinearlabs/client/integrationTests/OracleTests.java:38]: we have to wait for the server to receive the update here, don't we?
For example, we could store ServerProxys known updateId right before we send the command, then, in the 'done' callback we could require that the updateId actually changed AND the oracle flag is as expected.

File: projects/playground/src/http/ContentManager.cpp [hunk 3]
```diff
@@ -225,9 +225,11 @@ bool ContentManager::feedWebSocket(const tWebsocketConnection &c)
 
     if(c->getLastSentUpdateId() != currentUpdateId)
     {
-      DebugLevel::info("Updating websocket", c->getConnection(), "with document for updateID", currentUpdateId);
+      auto canOmitOracles = c->canOmitOracles(currentUpdateId);
+      DebugLevel::info("Updating websocket", c->getConnection(), "with document for updateID", currentUpdateId,
```
Comment (hhoegelo) [inline projects/playground/src/http/ContentManager.cpp:229]: is the info still needed?

File: projects/playground/src/device-settings/DirectLoadSetting.cpp [hunk 4]
```diff
@@ -8,11 +8,18 @@ DirectLoadSetting::DirectLoadSetting(Settings &settings)
 {
 }
 
-void DirectLoadSetting::load(const Glib::ustring &text)
+void DirectLoadSetting::load(const Glib::ustring &text, Initiator initiator)
 {
-  m_isLoading = true;
-  BooleanSetting::load(text);
-  m_isLoading = false;
+  if(initiator != Initiator::EXPLICIT_WEBUI)
```
Comment (hhoegelo) [inline projects/playground/src/device-settings/DirectLoadSetting.cpp:13]: Why is HWUI different from WebUI regarding this? Maybe it should be (initiator == Initiator::Load) instead? If not, why did you add the Initiator::Load?

File: projects/NonMaps/src/main/java/com/nonlinearlabs/client/integrationTests/OracleTests.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        server.setSetting("DirectLoad", "off");
+    }
+
+    private void selectPresetViaNewWebSocket() {
+        addStep(() -> {
+            documentReceived = false;
+            webSocket.send("/presets/banks/select-preset?uuid=" + findBank().getPreset(0).getUUID().toString() + "&isOracle=true");
+        }, () -> {
+            boolean wasOracleOfServer = server.lastDocumentCouldOmitOracles();
+            return !wasOracleOfServer && documentReceived;
```
Comment (hhoegelo) [inline projects/NonMaps/src/main/java/com/nonlinearlabs/client/integrationTests/OracleTests.java:50]: Hmm, I am not sure: You assume that the server should have received an update document, because the local WebSocket did. I don't know if this is necessarily true. The logic as applied in 'selectPresetViaProxy' looks more solid to me (except the 'documentReceived'-part which should be dispensable.

**Review by hhoegelo — APPROVED** [review]
